proposals/0652-import-shadowing.rst

```diff
@@ -265,36 +265,60 @@ something like
 then the ``foo`` exported by ``A`` should be the one defined in
 ``A``'s top-level.
 
-When modules are reexported wholesale, shadowing doesn't come into
-play, and so we keep the behaviour without this extension: the form
-``module M`` names the set of all entities that are in scope with both
-an unqualified name ``e`` and a qualified name ``M.e``. Example:
+.. _module-reexports:
 
-::
+Module reexports
+^^^^^^^^^^^^^^^^
 
- module A (module M) where
+In an export list, `Paragraph 5 of Section 5.2 of the Haskell 2010 report
+<https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_
+specifies that the form ``module M`` names the set of all entities
+that are in scope with both an unqualified name ``e`` and a qualified
+name ``M.e``.  So in this example, *without ``ImportShadowing``*:
 
- import M -- this exports "foo"
+::
 
- foo = ...
+ module A (module M) where
 
-Here, it is ``M.foo`` that is (re-)exported by ``A``, not ``A.foo``.
+ import M (foo, wombat)
+ import N (foo)
 
-If both ``module M`` and ``foo`` are exported, then that is a
-conflicting export error, and should be reported the same way as
-conflicts between exporting ``module M1`` and ``module M2`` without
-this extension. Example:
+ foo = True
 
-::
+the ``module M`` exports ``M.foo`` because that entity is in scope
+both as ``M.foo`` and with unqualified name ``foo``.  The fact that
+``N.foo`` and ``A.foo`` are *also* in scope with unqualified name
+``foo`` does not matter.
 
- module A (foo, module M) where
+With ``ImportShadowing``, however, the local definition of ``A.foo``
+*completely hides* the unqualified imports of ``M.foo`` and ``N.foo``;
+so now ``M.foo`` is no longer in scope with unqualified name ``foo``;
+so the ``module M`` export list exports only ``M.wombat``.
 
- import M -- this exports "foo"
+Conflicting exports
+^^^^^^^^^^^^^^^^^^^
 
- foo = ...
+Section 5.2 of the Haskell 2010 report
+`Section 5.2 of the Haskell 2010 report
+<https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_
+says "The unqualified names of the entities exported by a module must
+all be distinct".  That condition remains unchanged with this
+proposal. For example consider
 
-This should report a conflict between the export items ``foo``
-(resolving to ``A.foo``) and ``M.foo``.
+::
+  module A( module M, module N, foo ) where
+     import M( foo, wombat )
+     import N( wombat )
+    foo = True
+
+Without ``ImportShadowing`` the export list would attempt to export
+``M.foo`` and ``A.foo``; and would report a conflict. It also attempts
+to export ``M.wombat`` and ``N.wombat`` and would again report a
+conflict (assuming they are distinct entities).
+
+With ``ImportShadowing`` the ``wombat`` behaviour is unchanged; but
```

Comment:
Add a parenthesis:

...behaviour is unchanged (i.e. an export conflict is reported);